test/e2e/self_hosted_test.go

```diff
@@ -64,10 +64,9 @@ var _ = Describe("When testing Cluster API working on single-node self-hosted cl
 			BootstrapClusterProxy:    bootstrapClusterProxy,
 			ArtifactFolder:           artifactFolder,
 			SkipCleanup:              skipCleanup,
-			Flavor:                   "topology-single-node-cluster",
+			Flavor:                   "topology-single-kcp-cluster",
 			InfrastructureProvider:   ptr.To("docker"),
 			ControlPlaneMachineCount: ptr.To[int64](1),
```

Comment:
nit:

I would prefer keeping this and adding a comment that the template is not using the var.

Currently in the test it falls back to worker count 1